services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -796,14 +769,32 @@ void installDropLoopRule(DatapathId dpid) throws SwitchOperationException {
         }
     }
 
-    private long buildAndinstallMeter(final IOFSwitch sw, final DatapathId dpid, final long bandwidth,
-                                      final long burstSize, final long meterId)
-            throws OfInstallException {
-        logger.debug("installing meter {} on switch {} width bandwidth {}", meterId, dpid, bandwidth);
+    private void verifySwitchSupportsMeters(IOFSwitch sw) throws UnsupportedSwitchOperationException {
+        if (OVS_MANUFACTURER.equals(sw.getSwitchDescription().getManufacturerDescription())) {
+            throw new UnsupportedSwitchOperationException(sw.getId(),
+                    format("Meters are not supported on OVS switch %s", sw.getId()));
+        }
+
+        if (sw.getOFFactory().getVersion().compareTo(OF_12) <= 0) {
+            throw new UnsupportedSwitchOperationException(sw.getId(),
+                    format("Meters are not supported on the switch %s because of OF version %s",
+                            sw.getId(), sw.getOFFactory().getVersion()));
+        }
+    }
+
+    // FIXME: centec switches can't recognize PKTPS flag for meters.
+    // Need to simplify detection if the switch don't support PKTPS flag.
+    private boolean isSwitchSupportsPktpsFlag(IOFSwitch sw) {
+        return !StringUtils.contains(sw.getSwitchDescription().getManufacturerDescription(), "Centec");
+    }
+
+    private void buildAndInstallMeter(IOFSwitch sw, Set<OFMeterFlags> flags, long bandwidth, long burstSize,
+                                      long meterId) throws OfInstallException {
+        logger.info("Installing meter {} on switch {} width bandwidth {}", meterId, sw.getId(), bandwidth);
 
-        Set<OFMeterFlags> flags = new HashSet<>(asList(OFMeterFlags.KBPS, OFMeterFlags.BURST));
         OFFactory ofFactory = sw.getOFFactory();
 
+        // NB: some switches can't accept zero value for burst size
```

Comment:
... it is confusing. "Some switches replace 0 burst size value with some predefined value" should be less confusing..